sleap/io/format/nix.py

```diff
@@ -0,0 +1,402 @@
+        return SleapObjectType.misc
+
+    @property
+    def name(self) -> str:
+        """Human-reading name of the file format"""
+        return (
+            "NIX file flavoured for animal tracking data https://github.com/g-node/nix"
+        )
+
+    @classmethod
+    def can_read_file(cls, file: FileHandle) -> bool:
+        """Returns whether this adaptor can read this file."""
+        return False
+
+    def can_write_filename(self, filename: str) -> bool:
+        """Returns whether this adaptor can write format of this filename."""
+        return filename.endswith(tuple(self.all_exts))
+
+    @classmethod
+    def does_read(cls) -> bool:
+        """Returns whether this adaptor supports reading."""
+        return False
+
+    @classmethod
+    def does_write(cls) -> bool:
+        """Returns whether this adaptor supports writing."""
+        return True
+
+    @classmethod
+    def read(cls, file: FileHandle) -> object:
+        """Reads the file and returns the appropriate deserialized object."""
+        raise NotImplementedError("NixAdaptor does not support reading.")
+
+    @classmethod
+    def __check_video(cls, labels, video):
+        if video is None and len(labels.videos) == 0:
+            raise ValueError(
+                f"There are no videos in this project. NO output file will be be written."
+            )
+        if video is not None and video not in labels.videos:
+            raise ValueError(
+                f"Specified video {video} is not part of this project. NO output file will be be written."
+            )
+
+    @classmethod
+    def write(
+        cls,
+        filename: str,
+        source_object: object,
+        source_path: str = "None",
+        video: Video = None,
+    ):
+        """Writes the object to a file."""
+        cls.__check_video(source_object, video)
+
+        def create_file(filename, project, video):
+            print(f"...creating nix file {filename} for {project}", end="\t")
+            nf = nix.File.open(filename, nix.FileMode.Overwrite)
+            s = nf.create_section("TrackingAnalysis", "nix.tracking.metadata")
+            s["version"] = "0.1.0"
+            s["format"] = "nix.tracking"
+            s["definitions"] = "https://github.com/bendalab/nixtrack"
+            s["writer"] = str(cls)[8:-2]
+            if project is not None:
+                s["project"] = project
+
+            name = Path(video.backend.filename).name
+            b = nf.create_block(name, "nix.tracking_results")
+
+            # add video metadata, if exists
+            src = b.create_source(name, "nix.tracking.source.video")
+            sec = src.file.create_section(name, "nix.tracking.source.video.metadata")
+            sec["filename"] = video.backend.filename
+            sec["fps"] = getattr(video.backend, "fps", 0.0)
+            sec.props["fps"].unit = "Hz"
+            sec["frames"] = video.num_frames
+            sec["grayscale"] = video.backend.grayscale
```

Comment:
We are slowly integrating grayscale support into all `Video` backends (not all have this attribute yet):
```suggestion
            sec["grayscale"] = getattr(video.backend, "grayscale", None)
```